flow360/component/simulation/meshing_param/edge_params.py

```diff
@@ -54,12 +47,41 @@ class SurfaceEdgeRefinement(_BaseEdgeRefinement):
     """
 
     name: Optional[str] = pd.Field(None)
+    entities: Optional[EntityList[Edge]] = pd.Field(None, alias="edges")
+    # Note: Per edge `growth_rate` specification is actually not supported.
+    # Note: This can be only specified in global manner.
+    growth_rate: Optional[float] = pd.Field(
+        None,
+        description="Growth rate for surface mesh layers grown from edges. Default is 1.2",
+        ge=1,
+    )
     refinement_type: Literal["SurfaceEdgeRefinement"] = pd.Field(
         "SurfaceEdgeRefinement", frozen=True
     )
-    method: Union[
-        AngleBasedRefinement,
-        HeightBasedRefinement,
-        AspectRatioBasedRefinement,
-        ProjectAnisoSpacing,
-    ] = pd.Field(discriminator="type")
+    method: Optional[
+        Union[
+            AngleBasedRefinement,
+            HeightBasedRefinement,
+            AspectRatioBasedRefinement,
+            ProjectAnisoSpacing,
+        ]
+    ] = pd.Field(None, discriminator="type")
+
+    @pd.model_validator(mode="after")
+    def _check_valid_setting_combination(self):
+        """Check if the settings are valid in global or per-item context."""
+        if self.entities is not None:
+            # Is per-item edge refinement
+            if self.growth_rate is not None:
+                raise ValueError(
+                    "`growth_rate` can be only specified in global manner, not per edge."
+                )
+            if self.method is None:
+                raise ValueError("`method` must be specified for per-item surface edge refinement.")
+        else:
+            # Is Global edge refinement
+            if self.growth_rate is None:
+                self.growth_rate = 1.2  # Applying default
+            if self.method is not None:
+                raise ValueError("`method` can be only specified in per-item manner, not global.")
+        return self
```

Comment:
how about we move this validator to `refinements` validator which will be applied on refinements of type SurfaceEdgeRefinement. And then there is only one check we need to do:
check if set(growth rates) is length 1 (need to exclude None). If it is not, we can send warning instead of error that first appearance of growth_rate will be used.

Reply:
I see this relieves the requirement a bit but two issues:
1. If we have:
```python
[SurfaceEdgeRefinement(entities=[edge1, edge2], growth_rate = 1.4), SurfaceEdgeRefinement(growth_rate = 1.2)]
```
Then we can not say we use the 1st appearance? Since the eneities-is-none one is supposed to be global but is second in place?
2. Personally I think if user did not specify a global setting it might be arbitrary for us to tell them that we are going to use whatever is the first in the list? The result is going to change depending on the list ordering then.

Reply:
ok, so how about we raise error instead of warning and ask user to provide only one value?

Reply:
It is on todo list. See https://flow360.atlassian.net/browse/SCFD-2552. There are some values that does not require a global setting. For example max_edge_length.

Reply:
right. My problem is that now:
```python
[SurfaceEdgeRefinement(entities=[edge1, edge2], growth_rate = 1.4)]
```
will raise an error, although it is perfectly valid setting. We can use this 1.4 as global, it will affect only these two edges anyway, regardless if local or global.


What I am proposing it to just move this check one level above and make sure that only one value is used. It will be also very easy to remove this check if we decide to support it.

Reply:
The global SurfaceEdgeRefinement-> growth_rate does not only affect the listed/mentioned edges. Both meshers that we have will use this global setting for auto-detected edges that does not even exist in the geometry file. So we still require an explicit global `growth_rate`.  Source: @rundaji 

For other global setting fields. We cannot use arbitrary global values for the same reason that they make difference.

Reply:
I do appreciate your effort to reduce validation error due to capability limitation.

Reply:
I talked with Tengda this morning and the UI will mask most of these to make it harder to error out. See here: https://flow360.atlassian.net/browse/SCFD-2501#:~:text=Change%20of%20plan%3A%20We%20will%20have%20interface%20that%20aims%20to%20be%20future%2Dproof.

Reply:
I see. This is indeed the problem. So this design needs to stay. I goes this is the same problem with `curvature_resolution_angle` that is applied to all surfaces, not only mentioned ones?

How about boundary layer growth rate? This seems only be applied on mentioned patches?